consensus/p2p/proposal/proposal_stream.go

```diff
@@ -0,0 +1,62 @@
+package proposal
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/consensus/starknet"
+	"github.com/starknet-io/starknet-p2pspecs/p2p/proto/consensus/consensus"
+	"google.golang.org/protobuf/proto"
+)
+
+type proposalStream struct {
+	messages           map[uint64]*consensus.StreamMessage
+	transition         Transition
+	stateMachine       ProposalStateMachine
+	nextSequenceNumber uint64
+}
+
+func newSingleProposalStream(transition Transition) starknetProposalStream {
+	return &proposalStream{
+		messages:           make(map[uint64]*consensus.StreamMessage),
+		transition:         transition,
+		stateMachine:       transition.InitialState(),
+		nextSequenceNumber: 0,
+	}
+}
+
+func (s *proposalStream) OnStreamMessage(streamMessage *consensus.StreamMessage) (*starknet.Proposal, error) {
+	if s.nextSequenceNumber != streamMessage.SequenceNumber {
+		s.messages[streamMessage.SequenceNumber] = streamMessage
+		return nil, nil
+	}
+
+	exists := true
+	for exists {
```

Comment:
Why this approach instead of `break`/`return` when the execution path that is the one `exists` to true. Nothing against this one. 

Since we are having `exists` as a "top" level variable maybe we can give it a more detailed name such as `streamNextMessage`.